slangpy/tests/slangpy_tests/test_code_gen.py

```
@@ -1438,6 +1438,108 @@ def test_tensor_uses_entrypoint_args(device_type: spy.DeviceType):
+    cd = func.debug_build_call_data(tid=spy.call_id(), tensors=tensors, i=0, _result=out)
+    if device_type == spy.DeviceType.cuda:
+        # Heuristic fires: array of resources in binding -> PB.
+        assert cd.use_entrypoint_args is False
+    else:
+        # Other backends keep size-based selection (Vulkan small threshold may
+        # still force PB; D3D12 typically keeps EPA at this size).
+        threshold = device.info.limits.max_entry_point_uniform_size
+        # No CUDA-specific override on these targets, so behavior is purely
+        # threshold-driven. Don't pin a specific value here, just assert the
+        # call data built without errors.
+        assert cd is not None and threshold > 0
```

Ruff (0.15.18)

[warning] Assertion should be broken down into multiple parts

Break down assertion into multiple parts

(PT018)